libnd4j/include/ops/declarable/generic/images/resize_images.cpp

```diff
@@ -0,0 +1,129 @@
+                if(block.getIArguments()->size() == 3) { // full stack of args
+//                    height = I_ARG(0);
+//                    width = I_ARG(1);
+                    method = I_ARG(2);
+                }
+                else if (block.getIArguments()->size() == 2) {
+//                    height = I_ARG(0);
+//                    width = I_ARG(1);
+                }
+            }
+            bool preserveAspectRatio = false; // - default value
+            bool alignCorners = false;
+            if (block.getBArguments()->size()) {
+                alignCorners = B_ARG(0);
+                if (block.getBArguments()->size() > 1)
+                    preserveAspectRatio = B_ARG(1);
+            }
+
+            return helpers::resizeImagesFunctor(block.launchContext(), image, width, height, (helpers::ImageResizeMethods)method, alignCorners, output);
+        }
+
+        DECLARE_SHAPE_FN(resize_images) {
+            auto shapeList = SHAPELIST(); 
+            auto in = inputShape->at(0);
+
+            Nd4jLong* outputShape;
+
+            int width;
+            int height;
+            if (block.width() > 1) {
+                auto size = INPUT_VARIABLE(1);
+                REQUIRE_TRUE(size->lengthOf() == 2, 0, "resize_images: Resize params is a pair of values, not %lld.", size->lengthOf());
+                width = size->e<int>(1);
+                height = size->e<int>(0);
+            }
+            else {
+                REQUIRE_TRUE(block.getIArguments()->size() > 1 && block.getIArguments()->size() < 4, 0, "resize_images: Method and size should be given properly.");
+                if(block.getIArguments()->size() == 3) { // full stack of args
+                    height = I_ARG(0);
+                    width = I_ARG(1);
+                }
+                else if (block.getIArguments()->size() == 2) {
+                    height = I_ARG(0);
+                    width = I_ARG(1);
+                }
+            }
+
+            double ratio = shape::sizeAt(in, 1) / (0.0 + shape::sizeAt(in, 2));
+            if (block.getBArguments()->size() > 1) {
+                if (B_ARG(1)) {
+                    width = math::nd4j_ceil<double, int>(height / ratio);
+                }
+            }
+
+            ALLOCATE(outputShape, block.getWorkspace(), shape::shapeInfoLength(4), Nd4jLong);
+            outputShape[0] = 4;
+            outputShape[1] = in[1];
+            outputShape[2] = height;
+            outputShape[3] = width;
+            outputShape[4] = in[4];
+            ShapeUtils::updateStridesAndType(outputShape, DataType::FLOAT32, shape::order(in));
+
+            shapeList->push_back(CONSTANT(outputShape));
+            return shapeList;
+        }
+        DECLARE_TYPES(resize_images) {
+            getOpDescriptor()
+                    ->setAllowedInputTypes(0, {ALL_FLOATS, ALL_INTS})
+                    ->setAllowedInputTypes(1, {ALL_INTS})
+                    ->setAllowedOutputTypes({ALL_FLOATS});
```

Comment:
Same question. You have fixed FLOAT32 output in shape function, and allow ALL_FLOATS during validation. WTF?